server/src/main/java/org/elasticsearch/common/settings/LocallyMountedSecrets.java

```diff
@@ -191,20 +205,45 @@ LocalFileSecrets processSecretsFile(Path path) throws IOException {
         try (
             var fis = Files.newInputStream(path);
             var bis = new BufferedInputStream(fis);
-            var parser = JSON.xContent().createParser(XContentParserConfiguration.EMPTY, bis)
+            var parser = JSON.xContent()
+                .createParser(XContentParserConfiguration.EMPTY.withDeprecationHandler(DeprecationHandler.IGNORE_DEPRECATIONS), bis)
         ) {
             return secretsParser.apply(parser, null);
         }
     }
 
-    record LocalFileSecrets(Map<String, String> map, ReservedStateVersion metadata) implements Writeable {
+    record LocalFileSecrets(Map<String, byte[]> map, ReservedStateVersion metadata) implements Writeable {
```

Comment:
The name map seems far too generic. Can we rename this to something like KeyStoreWrapper uses, `entries`?